maven-agent/src/main/java/hudson/maven/agent/Main.java

```diff
@@ -195,8 +202,13 @@ public static void addPlexusComponents(URL[] modules) {
     /**
      * Called by the code in remoting to launch.
      */
-    public static int launch(String[] args) throws NoSuchMethodException, IllegalAccessException, NoSuchRealmException, InvocationTargetException, ClassNotFoundException {
+    public static int launch(String[] args)
```

Comment:
🐛 It breaks the compatibility of the public method. Ideally there should be a new one with better interface